### PR TITLE
Fix itilcategory dropdown in tickets/changes/problems

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Internal ref: 20757.

Given the following ITIL categories in multiple entities:
image

---

Before (categories in child entities are ignored):
image

---

After:
image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: inc/session.class.php [hunk 1]
```diff
@@ -1414,4 +1414,15 @@ static function mustChangePassword() {
...
+
+   /**
+    * Get active entities id.
+    *
+    * @since 9.5
+    *
+    * @return array
+    */
+   public static function getActiveEntities() {
+      return $_SESSION['glpiactiveentities'];
```
Comment (orthagh) [inline inc/session.class.php:1426]:
```suggestion
      return $_SESSION['glpiactiveentities'] ?? [];
```

**Review by trasher — APPROVED** [review]

**Comment by cedric-anne** [comment]
It means that you can create a ticket in entity A1 with a category of AA2. If you display the ticket from entity A1 without "tree structure", it will be linked to a category that you cannot access. Is it intended ?

**Comment by AdrienClairembault** [comment]
> It means that you can create a ticket in entity A1 with a category of AA2. If you display the ticket from entity A1 without "tree structure", it will be linked to a category that you cannot access. Is it intended ?

It doesn't seems to be a problem for me, in this case you can see the category but not interact with it so it should be fine ?

**Review by orthagh — CHANGES_REQUESTED** [review]
As seen together, i disagree strongly with this change.

This is not desirable to get the entity/is_recursive couple from current session when retrieving foreign key data.
We may have bugs in rules/notifications due to this

**Pull request closed without merging**